tools/benchmark_tool/openvino/tools/benchmark/utils/inputs_filling.py

```diff
@@ -158,111 +219,158 @@ def get_extension(file_path):
-    shape = info.shape
-    images = np.ndarray(shape)
-    if from_map:
-        image_index = request_id * batch_size
-    else:
-        image_index = request_id * batch_size * input_size + input_id
-
-    scale_mean = (not np.array_equal(info.scale, (1.0, 1.0, 1.0)) or not np.array_equal(info.mean, (0.0, 0.0, 0.0)))
-
-    for b in range(batch_size):
-        image_index %= len(image_paths)
-        image_filename = image_paths[image_index]
-        logger.info(f'Prepare image {image_filename}')
-        image = cv2.imread(image_filename)
-        new_im_size = tuple((info.width, info.height))
-        if image.shape[:-1] != new_im_size:
-            logger.warning(f"Image is resized from ({image.shape[:-1]}) to ({new_im_size})")
-            image = cv2.resize(image, new_im_size)
-
-        if scale_mean:
-            blue, green, red = cv2.split(image)
-            blue = np.subtract(blue, info.mean[0])
-            blue = np.divide(blue, info.scale[0])
-            green = np.subtract(green, info.mean[1])
-            green = np.divide(green, info.scale[1])
-            red = np.subtract(red, info.mean[2])
-            red = np.divide(red, info.scale[2])
-            image = cv2.merge([blue, green, red])
-
-        if info.layout in ['NCHW', 'CHW']:
-            image = image.transpose((2, 0, 1))
-
-        images[b] = image
-
-        if from_map:
+def get_image_tensors(image_paths, info, batch_sizes):
+    processed_frames = 0
+    widthes = info.widthes if info.is_dynamic else [len(info.width)]
+    heights = info.heights if info.is_dynamic else [len(info.height)]
+    tensors = []
+    process_with_original_shapes = False
+    num_shapes = len(info.shapes)
+    if num_shapes == 0:
+        process_with_original_shapes = True
+    num_images = len(image_paths)
+    niter = max(num_shapes, num_images)
+    for i in range(niter):
+        shape = list(info.shapes[i % num_shapes]) if num_shapes else []
+        images = np.ndarray(shape=shape, dtype=np.uint8)
+        image_index = processed_frames
+        scale_mean = (not np.array_equal(info.scale, (1.0, 1.0, 1.0)) or not np.array_equal(info.mean, (0.0, 0.0, 0.0)))
+
+        current_batch_size = 1 if process_with_original_shapes else batch_sizes[i % num_shapes]
+        for b in range(current_batch_size):
+            image_index %= num_images
+            image_filename = image_paths[image_index]
+            logger.info(f'Prepare image {image_filename}')
+            image = cv2.imread(image_filename)
+            if process_with_original_shapes:
+                logger.info(f'Image will be processed with original shape - {image.shape[:-1]}')
+            else:
+                new_im_size = (widthes[i % num_shapes], heights[i % num_shapes])
+                if image.shape[:-1] != new_im_size:
+                    logger.warning(f"Image is resized from ({image.shape[:-1]}) to ({new_im_size})")
+                    image = cv2.resize(image, new_im_size)
+
+            if scale_mean:
```

Comment:
Are there any plans to cover moving preprocessing to the OV? I suppose, setting layouts, means/scales should be done via OV API when it will be available.